src/renderers/dom/client/wrappers/ReactDOMSelect.js

```diff
@@ -58,6 +58,28 @@ function warnIfValueIsNull(props) {
   }
 }
 
+function warnIfDuplicateValues(inst) {
+  const options = inst._currentElement.props.children;
+
+  if (options === undefined) {
+    return;
+  }
+
+  // Displays a warning for all duplicate values in select element. Does not exit after first duplicate is found.
+  for (var i = 0; i < options.length-1; i++) {
+    if (options[i].props.value !== undefined) {
+      for (var j = i + 1; j < options.length; j++) {
+        if (options[i].props.value === options[j].props.value) {
+          warning(
+            false,
+            `Select element contains duplicate option value ${options[i].props.value} in options #${i} & #${j}`
```

Comment:
We try to include a stack trace for new warnings.  Again, https://github.com/facebook/react/pull/7040 is a good example.